torch/csrc/distributed/autograd/context/dist_autograd_context.cpp

```diff
@@ -12,6 +12,18 @@ int64_t DistAutogradContext::context_id() const {
   return context_id_;
 }
 
+DistAutogradContext::known_worker_set_type DistAutogradContext::
+    getKnownWorkerIds() const {
+  return knownWorkerIDs_;
+};
+
+void DistAutogradContext::addKnownWorkerID(const rpc::WorkerId& workerId) {
+  std::lock_guard<std::mutex> guard(lock_);
+
+  auto id = workerId.id_;
```

Comment:
I guess as a minor performance optimization, we can read the id outside of the lock

Reply:
For now, I think its simpler to just do `knownWorkerIds_.set(workerId.id_)`.